src/js/test/unit/scheduler.test.ts

```diff
@@ -1,18 +1,29 @@
-import * as chai from "chai";
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
 import { scheduleCallback } from "../../scheduler";
 
 describe("scheduleCallback", () => {
-  it("should call the callback immediately if timeout is 0", () => {
-    const start = Date.now();
+  it("should call the callback immediately if timeout is 0", (t) => {
```

Comment:
This test passed on my laptop.
It also passed on CI, but after the test finished, the callback seemed to be run and the assertion failed only on CI.

Since the behavior depends on timing, it isn’t deterministic. To make it deterministic, I mocked setImmediate and setTimeout, which are invoked inside scheduleCallback.

I hope this will make the tests run deterministically.